### DISCUSSION TITLE
Hybrid Configuration (NvChad + CocNvim + VSCode)

### DISCUSSION THREAD

**Discussion by martin-braun** [vscode-neovim/vscode-neovim#1239] [Show and tell]
I ditched VSCode in the past and picked up https://github.com/NvChad/NvChad for performance reasons. However I disabled some default plugins and switched the highlighting, code suggestions and LSP to https://github.com/neoclide/coc.nvim which works pretty good for the most times. Now I found this (and I like it for obvious reasons), but I don't want to give up on a terminal solution most of the times. I love working in `tmux`, but sometimes VSCode would be helpful (I don't like debugging with a DAP within Neovim, tbh).

I took my configuration and ported it to VSCode (while disabling most plugins in that mode, though). I substituted all my mappings in VSCode and also provide a keybindings.json that works with it (when focus is not on the text editor). Now I can choose Terminal or VSCode mode and it works the same for me. I also added some bindings to substitute `tmux` behavior (prefix is `^T`) in a minimal sense. It's handy when I need a terminal below a fullscreen VSCode.

https://github.com/martin-braun/CocChad 
(BEWARE: `./link` will remove all your Neovim settings and start fresh)

I just post it here to give you an idea or inspire you, use at your own risk.

**Comment by agungwas** [comment]
just wondering, is this best way to neovim beginner to move from vs-code?

**Reply by ClaudioDavi** [reply]
@agungwas actually, I'd say that  its not. The best way is to use the vim extension on vscode until you get the hang of it, then once you are confortable you move to a more complete solution like nvchad or build your own like Kickstart.nvim

**Reply by martin-braun** [reply]
I agree, it's not the best way, it works for me and I shared it to inspire you. It's just an example on how to write a configuration that works in VSCode and Neovim, but it's been a while and I haven't used VSCode in a while. My configuration also runs on an older NvChad, I want to rice my own in the future.

**Comment by theol0403** [comment]
LazyVim also has first-class support for vscode-neovim, so both will work out of the box with a LazyVim config
